aztec-up/bin/aztec

```diff
@@ -53,11 +53,205 @@ function get_env_vars {
     awk -F"'" '{for(i=2;i<=NF;i+=2) printf $i " "}'
 }
 
+function validate_working_directory {
+  # Validate working directory is under $HOME for Docker containerization
+  if [[ $PWD != ${HOME}* ]]; then
+    >&2 echo "Due to how we containerize our applications, we require your working directory to be somewhere within $HOME."
+    exit 1
+  fi
+}
+
+function run_nargo {
+  # Helper function to run nargo commands with proper Docker setup
+  # Usage: run_nargo <command> [args...]
+
+  # Determine if we need interactive/tty flags
+  if [ -t 0 ]; then
+    if [ -t 1 ]; then
+      TTY_FLAGS="-ti"
+    else
+      TTY_FLAGS="-i"
+    fi
+  fi
+
+  # Run nargo via docker
+  docker run ${TTY_FLAGS:-} \
+    --user $(id -u):$(id -g) \
+    -v $HOME:$HOME \
+    -e HOME=$HOME \
+    --workdir="$PWD" \
+    --entrypoint=/usr/src/noir/noir-repo/target/release/nargo \
+    $DOCKER_REPO:$VERSION "$@"
+}
+
+function setup_aztec_contract_project {
```

Comment:
Added this function and `parse_project_flags_and_build_nargo_args` because the functionality is shared between `nargo new` and `nargo init` commands.